### PR TITLE
Fix #185858: Add option to enable/disable editor group maximization on double clicking tab title

### LINKED ISSUE #185858: Allow to disable editor group maximize on double click
Occasionally a mouse click turns into a double click (which is hard to avoid), but I don't want to maximize the editing group and have to get it back every time I accidentally maximize it.

I can't find the relevant settings in the Settings, I hope this feature can be turned off, thank you very much!

### PR DESCRIPTION
**Pull request by hsfzxjy**
This PR fixes issue #185858. It adds an option to enable or disable the behavior that double clicking a tab's title would maximize the current editor group.

### CODE REVIEW

**Comment by hsfzxjy** [comment]
@microsoft-github-policy-service agree

**Comment by starball5** [comment]
For the naming "toggleGroupMaximizationOnDoubleClick", I might suggest "toggleGroupMaximizationOnDoubleClickingTab"

**Comment by hsfzxjy** [comment]
@starball5 Fair enough. I've changed the naming.

**Review by bpasero — CHANGES_REQUESTED** [review]
LGTM but maybe the setting name could be renamed. Now it is `toggleGroupMaximizationOnDoubleClickingTab`, other settings tend to put the action first, also the command is called "Toggle Editor Group Sizes" so maybe `doubleClickTabToToggleEditorGroupSizes`?

**Comment by hsfzxjy** [comment]
Sure. I fixed it.

File: src/vs/workbench/browser/workbench.contribution.ts [hunk 1]
```diff
@@ -282,6 +282,11 @@ const registry = Registry.as<IConfigurationRegistry>(ConfigurationExtensions.Con
 				'default': false,
 				'description': localize('centeredLayoutDynamicWidth', "Controls whether the centered layout tries to maintain constant width when the window is resized.")
 			},
+			'workbench.editor.doubleClickTabToToggleEditorGroupSizes': {
+				'type': 'boolean',
+				'default': true,
+				'description': localize('doubleClickTabToToggleEditorGroupSizes', 'Controls whether to maximize/restore the size of current editor group when double clicking on a tab\' title.'),
```
Comment (bpasero) [inline src/vs/workbench/browser/workbench.contribution.ts:288]:
```suggestion
				'markdownDescription': localize({ comment: ['This is the description for a setting. Values surrounded by single quotes are not to be translated.'], key: 'doubleClickTabToToggleEditorGroupSizes' }, "Controls whether to maximize/restore the editor group when double clicking on a tab. This value is ignored when `#workbench.editor.showTabs#` is disabled.")
```

**Review by bpasero — CHANGES_REQUESTED** [review]
Almost 👍

**Review by bpasero — APPROVED** [review]

**Review by alexr00 — APPROVED** [review]

**Pull request merged**